### PR TITLE
CORE-6781

### PR DESCRIPTION
**Pull request by MrMatthewLayton**
Added CPK Constraint Interfaces and Tests
**NOTE : PR AND MERGE CORE-6779 BEFORE THIS ONE!**

### CODE REVIEW

**Review by yarivus — APPROVED** [review]
LGTM

File: ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/Constraints.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.cipher.suite.DigestService
+
+/**
+ * Defines a constraint that determines which contract-code-containing CPK can be used with a [Contract].
+ */
+@CordaSerializable
+@DoNotImplement
+interface CpkConstraint {
```
Comment (mnesbit) [inline ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/Constraints.kt:12]: I don't think we need to expose cpkconstraints in C5. I think the automatic sandboxing and AMQP rules will mean that this is enforced by the platform and not user modifiable so delete.

**Review by mnesbit — COMMENTED** [review]
I think this PR will end up deleted

**Review by yarivus — COMMENTED** [review]
LGTM

**Review by yarivus — APPROVED** [review]
LGTM

**Pull request closed without merging**